models/points.go

```diff
@@ -2088,16 +2092,43 @@ func CompareTags(a, b Tags) int {
 
 // Get returns the value for a key.
 func (a Tags) Get(key []byte) []byte {
-	// OPTIMIZE: Use sort.Search if tagset is large.
-
-	for _, t := range a {
-		if bytes.Equal(t.Key, key) {
-			return t.Value
+	if len(a) < 25 {
```

Comment:
Whilst I'm always happy to see `TODO`s being sorted out, adding a binary search here concerns me a little bit unless it's accompanied by an audit of the codebases.

What I'm mainly worried about is that Tags are being created that are not in order. I assume it wouldn't be too hard to verify the API is being used correctly of course.

Reply:
Moved this commit to #13582 to complete audit of `models.Tags` usage and fix any issues related #13581